### PR TITLE
Add "Julia interface" to interface.md

### PR DESCRIPTION
**Pull request by singularitti**
Greetings,
As a user of spglib and an active contributor within the Julia community, I have successfully ported the C-API to Julia, which can be accessed [here](https://github.com/singularitti/Spglib.jl). However, the magnetic interface is yet to be implemented and I would greatly appreciate any assistance in this regard.
The Spglib.jl is already being utilized by some individuals, and it is my aspiration to further promote both spglib and Spglib.jl, thereby expanding their user base.

### CODE REVIEW

File: doc/interface.md [hunk 1]
```diff
@@ -19,6 +19,11 @@ obsolete and is not recommended to use.
 Rust interface contributed by Seaton Ullberg. The documentation is
 found at <https://docs.rs/crate/spglib/>.
 
+## Julia interface
+
+Julia interface contributed by [Qi Zhang](https://github.com/singularitti). The documentation is
```
Comment (LecrisUT) [inline doc/interface.md]: Can you link to the github repository or the package manager? Not sure about Julia packaging.

Comment (singularitti) [reply doc/interface.md]: Sure. My repo is here: https://github.com/singularitti/Spglib.jl.
But I put the link to JuliaHub: https://juliahub.com/ui/Packages/General/Spglib,
JuliaHub is kind of like Python's PyPI.

**Review by LecrisUT — APPROVED** [review]

**Comment by LecrisUT** [comment]
LGTM @atztogo @lan496 thoughts?

On my side I only have a few CI questions outside of this PR:
- how do you keep track of new releases? And where does the project define how the cmake is built?
- is it possible to run the CI on spglib main so that we can see if there are breaking changes to report? Maybe if you can make a reusable workflow that can be imported and run here?

**Comment by singularitti** [comment]
I'm not the expert but I will try to answer these questions:

> how do you keep track of new releases? And where does the project define how the cmake is built?

Julia has a mechanism called "artifacts", which ships binary executables or libraries. In the building package of that artifact (in this case, the package [`spglib_jll.jl`](https://github.com/JuliaBinaryWrappers/spglib_jll.jl)), [almost all platforms are supported](https://github.com/JuliaBinaryWrappers/spglib_jll.jl?tab=readme-ov-file#platforms) uniformly. So in any platform that supports Julia, you could use `spglib_jll` with the same API without building it by yourself.

But `spglib_jll` is just a plain Julia wrapper that wraps around the dynamic library file, it does not provide any native data structure and function support for Julia. That's why I wrapped it again to be a useful package [`Spglib.jl`](https://github.com/singularitti/Spglib.jl).

To keep track of the latest release of `spglib`, whenever there is a new release of `spglib`, we ship a new version of `spglib_jll`, with the same version number plus a suffix, like [`v2.0.2+0`](https://github.com/JuliaBinaryWrappers/spglib_jll.jl/releases).
The building script is [here](https://github.com/JuliaPackaging/Yggdrasil/blob/115ab389d388b2e97fbe17973929ec11c7c39183/S/spglib/build_tarballs.jl#L15-L26).
We use Docker containers to provide building environments to debug locally, and once it is finished, we push it to GitHub and let CI build the `spglib_jll` package.

However, I do have problems building the v2.1.0 version of `spglib`, I do not know which part of the [script](https://github.com/JuliaPackaging/Yggdrasil/blob/115ab389d388b2e97fbe17973929ec11c7c39183/S/spglib/build_tarballs.jl#L15-L26) is wrong since it was successful in the past.
The [build errors](https://buildkite.com/julialang/yggdrasil/builds/5031#018a80fa-2d72-47f2-af12-1a9274d37cad) are here.
I guess you changed some dependencies in this version? Would you mind telling me what might affect the building process? The changelog from v2.0.2 to v2.1.0 is too long and I cannot go through it immediately.

> is it possible to run the CI on spglib main so that we can see if there are breaking changes to report? Maybe if you can make a reusable workflow that can be imported and run here?

I'm not sure about what you mean here. I currently put all [tests](https://github.com/singularitti/Spglib.jl/tree/321f64f899788a68040389ddc9c13d12b7e641fb/test) here. Most of the CI builds are successful. Here is one [example](https://github.com/singularitti/Spglib.jl/actions/runs/6212866285/job/16863420169) (The errors are intentional since the package only works on Julia v1.6+.

**Review by lan496 — APPROVED** [review]
LGTM!

**Comment by LecrisUT** [comment]
I've responded in the julia PR about the packaging. But otherwise it seems the updates are opened manually? It is preferable that there is some automation for the release also for `rc` when possible.

> > is it possible to run the CI on spglib main so that we can see if there are breaking changes to report? Maybe if you can make a reusable workflow that can be imported and run here?
> 
> I'm not sure about what you mean here. I currently put all [tests](https://github.com/singularitti/Spglib.jl/tree/321f64f899788a68040389ddc9c13d12b7e641fb/test) here. Most of the CI builds are successful. Here is one [example](https://github.com/singularitti/Spglib.jl/actions/runs/6212866285/job/16863420169) (The errors are intentional since the package only works on Julia v1.6+.

I mean, can you make a github workflow that builds `spglib_jll` and than triggers the tests in `Spglib.jl` linked to that? We can then try out a design where your workflow is called within this repo. That way we are also made aware of any potential errors downstream. We currently have this for Fedora and partially for conda through release candidates.

**Comment by atztogo** [comment]
Thanks @singularitti. It is very nice to inform the existence of the Julia interface from the documentation officially.

**Review by atztogo — APPROVED** [review]

**Comment by singularitti** [comment]
Thank you very much! I'll look into the `spglib_jll` build errors.

**Comment by singularitti** [comment]
Sorry, would you mind changing the title to "Rust, Fortran, Julia, and Ruby interfaces"? I forgot to update that.

**Comment by LecrisUT** [comment]
I think we'll refactor that to something else to include packaging as well

**Comment by singularitti** [comment]
Cool! Please inform me if there's anything I can help.

**Pull request merged**